src/webgpu/api/validation/render_pass/render_pass_descriptor.spec.ts

```diff
@@ -713,3 +713,35 @@ g.test('multisample_render_target_formats_support_resolve')
       colorAttachments: [colorAttachment],
     });
   });
+
+g.test('timestamp_writes_location')
+  .desc('Test that entries in timestampWrites do not have the same location.')
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase(['timestamp-query']);
+  })
+  .fn(async t => {
+    const querySet = t.device.createQuerySet({
+      type: 'timestamp',
+      count: 1,
+    });
+
+    const timestampWriteA = {
+      querySet,
+      queryIndex: 0,
+      location: 'beginning' as GPURenderPassTimestampLocation,
+    };
+
+    const timestampWriteB = {
+      querySet,
+      queryIndex: 1,
+      location: 'beginning' as GPURenderPassTimestampLocation,
+    };
+
+    const colorTexture = t.createTexture({ format: 'rgba8unorm' });
+    const descriptor = {
+      colorAttachments: [t.getColorAttachment(colorTexture)],
+      timestampWrites: [timestampWriteA, timestampWriteB],
+    };
+
+    t.tryRenderPass(false, descriptor);
```

Comment:
Right now this always fails so it's hard to tell that the error is due to this specific error. The test should be parameterized so that it has at least one "control case" (where the validation passes), and it should also test what happens when both are `'end'`. So I'd test all 4 combinations:
```ts
  .paramsSubcasesOnly(u =>
    u //
      .combine('locationA', ['beginning', 'end'] as const)
      .combine('locationB', ['beginning', 'end'] as const)
  )
```
updating the rest of the test as necessary.

Reply:
The test is passing even though the locations are the same. I guess that Chromium doesn't generate a validation error.  

I add the combinations to the test. BTW, can we use .param instead of  .paramsSubcasesOnly in order to know what combinations fail more clear.

Reply:
Yup, that would be fine!